django/db/models/query_utils.py

```diff
@@ -91,6 +91,7 @@ def resolve_expression(
             split_subq=False,
             check_filterable=False,
             summarize=summarize,
+            reuse_with_filtered_relation=True,
```

Comment:
This seems wrong as it will now make any `Q` resolving take the `SQLQuery.join` path dedicated to dealing with filtered relations.

https://github.com/django/django/blob/01ae9d4ca9afdaf30a247e10e8333261a7d8224c/django/db/models/sql/query.py#L997-L1016